cloud/amazon/ec2_elb_lb.py

```diff
@@ -698,6 +710,28 @@ def _set_cross_az_load_balancing(self):
         self.elb_conn.modify_lb_attribute(self.name, 'CrossZoneLoadBalancing',
                                           attributes.cross_zone_load_balancing.enabled)
 
+    def _set_access_log(self):
+        attributes = self.elb.get_attributes()
+        if self.access_logs:
+            access_log = boto.ec2.elb.attributes.AccessLogAttribute()
+            access_log.enabled = True
+            access_log.s3_bucket_prefix = ''
+            access_log.emit_interval = 60
+            if 's3_location' not in self.access_logs:
+              self.module.fail_json(msg='s3_location information required')
+            access_log.s3_bucket_name = self.access_logs['s3_location']
+            if 's3_prefix' in self.access_logs:
+              access_log.s3_bucket_prefix = self.access_logs['s3_prefix']
+            if 'interval' in self.access_logs:
+              access_log.emit_interval = self.access_logs['interval']
+            if access_log.__dict__ != attributes.access_log.__dict__:
```

Comment:
This works better although I still think it's not ideal. If one or the other object accumulates extra data attributes stored in `__dict__` we might end up in a situation where the comparison always returns false and the module always thinks there has been a change.

Consider that in the code above we are explicitly walking over each attribute that's meaningfully shared across both object. It might make sense to just check if one of those has changed while we're doing that check.

Checkout  `__set_health_check` for a way you could approach it.